addon/lib/shot.js

```diff
@@ -198,11 +198,12 @@ class AbstractShot {
     if (attrs.clips) {
       for (let clipId in attrs.clips) {
         let clip = attrs.clips[clipId];
-        this.clips[clipId] = new this.Clip(this, clipId, clip);
+        this._clips[clipId] = new this.Clip(this, clipId, clip);
       }
     }
+
     for (let attr in attrs) {
-      if (attr != "clips" && this.REGULAR_ATTRS.indexOf(attr) == -1) {
+      if (attr !== "clips" && attr !== "id" && this.REGULAR_ATTRS.indexOf(attr) == -1) {
```

Comment:
The id shouldn't show up here, as it's a separate argument to the constructor and isn't mutable.  Is it getting through somewhere?

I guess maybe it could be okay if it does tend to arise naturally, though then we should assert that `attrs.id === this.id`